### ISSUE TITLE
Do we need rhdp theme JS files?

### ISSUE THREAD

**Issue by jordanpagewhite** [redhat-developer/rhd-frontend#128]
The following JS files currently reside in `/web/themes/custom/rhdp/js/` with the redhat-developer/developers.redhat.com repo/directory tree:

* all.min.js
* angular-sanitize.min.js
* angular.min.js
* devnationlive.json
* drupal-user-login-form.js
* highlightjs-init.js
* menu.js

Could we move all of these files into our redhat-developer/rhd-frontend project?

`drupal-user-login-form.js` and `highlightjs-init.js` need to be exported as individual dist assets, as they will only be loaded if their respective component(s) are loaded on a page/request in Drupal.

Do we still use `devnationlive.json`, or is this legacy waste?

Do we still need to load Angular? If so, for which component(s) and why? We are also loading Angular via a CDN here:

```
https://cdnjs.cloudflare.com/ajax/libs/angular.js/1.3.3/angular.min.js: { type: external, minified: true }
https://cdnjs.cloudflare.com/ajax/libs/angular-sanitize/1.3.3/angular-sanitize.min.js: { type: external, minified: true }
```

`all.min.js` It seems as though this file just contains FontAwesome data. Is that accurate? If so, will we abandon this file in favor of loading FontAwesome via a CDN?

**Comment by robpblake** [comment]
@jordanpagewhite Feels like this is a good time to audit what is actually in use and kill it with 🔥 if it's not being used.

I suspect a lot of the above has simply been bought into each update of the theme because no one was sure if it was in use.

**Comment by robpblake** [comment]
@jordanpagewhite I wonder if it's worth breaking this out into some smaller issues and whether this is actually better placed on the Drupal project?

For me I'm seeing the following issues:

1. What is using Angular JS (if anything) and why are we loading it both from a CDN and locally?
2. `all.js` needs to die so how do we make that happen? - We cannot continue to host a 3MB JS file. There needs to be a re-think of how things are working if this is a requirement
3. What is `devnationlive.json` and is it still needed?
4. What is `menu.js` and is it still needed?

I think if we break them down then they can be more easily tackled e.g. I could take a shot at simply removing Angular and seeing what breaks.

What do you think?

cheers,

Rob

**Comment by jordanpagewhite** [comment]
1. @robpblake I have started a thread in Slack moments ago to discuss removing the Help page/section from our site (currently in the top nav). Joe responded that he would be removing the Help item from the top nav later today, and that we would need to determine where to redirect that section. After we set up a redirect though, he said that we could remove the page. I believe that is the only page on the site that uses Angular but hopefully @KamiQuasi @adelasofia or @alistairmc can confirm that. It would be great to eliminate that dependency entirely if possible.
2. `all.min.js` was a FontAwesome file. I am not sure why an icon library is so incredibly huge. Either way, I have already commented out this file in the `rhdp2.libraries.yml` file in the `surge` branch. We should verify that, given the new way we are including FA as a dependency, we aren't just pulling in many MBs of content that we won't even be using.
3. This looks like some old content that was consumed by a past DevNation event. Perhaps @KamiQuasi or @adelasofia can confirm. I see that `rhdp-apps/devnationlive/index.html` makes a `fetch()` call to it.
4. `menu.js` is some jQuery for the menu, including mobile behavior (toggle, etc.), the search toggle, a slider for the topic nav for smaller viewports etc. It is only 46kb, so it seems like a very small concern now relative to the overall asset bloat.

Depending on whether or not @KamiQuasi and/or @adelasofia are aware of the purpose of the `devnationlive.json` file and whether or not we still need it, we might just be able to entirely delete Angular as a dependency, the `all.min.js` file and `devnationlive.json` possibly as early as tomorrow. I will wait until tomorrow. If there is more work, besides just deleting or leaving them as-is, for any of these assets, I will create individual tickets to track that work.

Thanks Rob.
